kotlinx-coroutines-core/common/src/channels/ArrayChannel.kt

```diff
@@ -41,7 +43,7 @@ internal open class ArrayChannel<E>(
     protected final override val isBufferAlwaysEmpty: Boolean get() = false
```

Comment:
Looks like this property should be called `hashElementBuffer`

Reply:
It is named this way to name the code like this easier to read (node, this is all done only as an optimization):
```
    protected open fun enqueueReceiveInternal(receive: Receive<E>): Boolean = if (isBufferAlwaysEmpty)
        queue.addLastIfPrev(receive) { it !is Send } else
        queue.addLastIfPrevAndIf(receive, { it !is Send }, { isBufferEmpty })
```